pkgs/development/haskell-modules/HACKING.md

```diff
@@ -0,0 +1,229 @@
+
+This build report can be fetched and re-generated for new Hydra evaluations.
+It may help contributors to try to keep the GitHub comment updated with the
+most recent build report.
+
+Maintainers should be given at least 7 days to fix up their packages when they
+break.  If maintainers don't fix up their packages with 7 days, then they
+may be marked broken before merging `haskell-updates` into `master`.
+
+#### Fix Broken Packages
+
+After getting the build report, you can see which packages and Hydra jobs are
+failing to build.  The most important jobs are the `maintained` and `mergeable`
+jobs. These are both defined in
+[`release-haskell.nix`](../../top-level/release-haskell.nix).
+
+`mergeable` is a set of the most important Haskell packages, including things
+like Pandoc and XMonad.  These packages are widely used.  We would like to
+always keep these building.
+
+`maintained` is a set of Haskell packages that have maintainers in Nixpkgs.
+We should be proactive in working with maintainers to keep their packages
+building.
+
+Steps to fix Haskell packages that are failing to build is out of scope for
+this document, but it usually requires fixing up dependencies that are now
+out-of-bounds.
+
+#### Mark Broken Packages
+
+Packages that do not get fixed can be marked broken with the following
+commands.  First check which packages are broken:
+
+```console
+$ maintainers/scripts/haskell/hydra-report.hs get-report
+$ maintainers/scripts/haskell/hydra-report.hs mark-broken-list
+```
+
+This shows a list of packages that reported broken on `x86_64-linux` on Hydra.
+
+Next, run the following command:
+
+```console
+$ maintainers/scripts/haskell/mark-broken.sh --do-commit
+```
+
+This first opens up an editor with the broken package list.  Some of these
+packages may have a maintainer in Nixpkgs.  If these maintainers have not been
+given 7 days to fix up their package, then make sure to remove those packages
+from the list before continuing.  After saving and exiting the editor, the
+following will happen:
+
+- packages from the list will be added to
+  [`configuration-hackage2nix/broken.yaml`](configuration-hackage2nix/broken.yaml)
+- `hackage-packages.nix` will be regenerated
+- the
+  [`configuration-hackage2nix/transitive-broken.yaml`](configuration-hackage2nix/transitive-broken.yaml)
+  file will be updated
+- `hackage-packages.nix` will be regenerated again
+- everything will be committed
+
+### Merge `haskell-updates` into `master`
+
+Now it is time to merge the `haskell-updates` PR you opened above.
+
+Before doing this, make sure of the following:
+
+- All Haskell packages that fail to build are correctly marked broken or
+  transitively broken.
+- The `maintained` and `mergeable` jobs are passing on Hydra.
+- The maintainers for any maintained Haskell packages that are newly broken
+  have been pinged on GitHub and given at least a week to fix their packages.
+  This is especially important for widely-used packages like `cachix`.
+
```

Comment:
As per yesterday's discussion, we should make it a point here to merge `master` into `haskell-updates` and waiting for the rebuilds to finish before merging the PR.

Reply:
Especially since I realized: Merging in master and hitting "eval" on hydra and seeing if anything new happens is really only a matter of a few minutes. I think we can afford to do that on every merge.

Reply:
I've rearranged this in adfec8b5e04.  Hopefully it is somewhat more clear now.